examples/pydantic_ai_ag_ui_examples/api/agentic_generative_ui.py

```diff
@@ -0,0 +1,142 @@
+        default=StepStatus.PENDING,
+        description='The status of the step (e.g., pending, completed)',
+    )
+
+
+class Plan(BaseModel):
+    """Represents a plan with multiple steps."""
+
+    steps: list[Step] = Field(
+        default_factory=lambda: list[Step](), description='The steps in the plan'
+    )
+
+
+class JSONPatchOp(BaseModel):
+    """A class representing a JSON Patch operation (RFC 6902)."""
+
+    op: Literal['add', 'remove', 'replace', 'move', 'copy', 'test'] = Field(
+        ...,
+        description='The operation to perform: add, remove, replace, move, copy, or test',
+    )
+    path: str = Field(..., description='JSON Pointer (RFC 6901) to the target location')
+    value: Any = Field(
+        default=None,
+        description='The value to apply (for add, replace operations)',
+    )
+    from_: str | None = Field(
+        default=None,
+        alias='from',
+        description='Source path (for move, copy operations)',
+    )
+
+
+@agui.agent.tool_plain
+def create_plan(steps: list[str]) -> StateSnapshotEvent:
+    """Create a plan with multiple steps.
+
+    Args:
+        steps: List of step descriptions to create the plan.
+
+    Returns:
+        StateSnapshotEvent containing the initial state of the steps.
+    """
+    plan: Plan = Plan(
+        steps=[Step(description=step) for step in steps],
+    )
+    return StateSnapshotEvent(
+        type=EventType.STATE_SNAPSHOT,
+        snapshot=plan.model_dump(),
+    )
+
+
+@agui.agent.tool_plain
+def update_plan_step(
+    index: int, description: str | None = None, status: StepStatus | None = None
+) -> StateDeltaEvent:
+    """Update the plan with new steps or changes.
+
+    Args:
+        index: The index of the step to update.
+        description: The new description for the step.
+        status: The new status for the step.
+
+    Returns:
+        StateDeltaEvent containing the changes made to the plan.
+    """
+    changes: list[JSONPatchOp] = []
+    if description is not None:
+        changes.append(
+            JSONPatchOp(
+                op='replace', path=f'/steps/{index}/description', value=description
+            )
+        )
+    if status is not None:
+        changes.append(
+            JSONPatchOp(op='replace', path=f'/steps/{index}/status', value=status.value)
+        )
+    return StateDeltaEvent(
+        type=EventType.STATE_DELTA,
+        delta=changes,
+    )
+
+
+@router.post('')
+async def handler(
+    input_data: RunAgentInput, accept: Annotated[str, Header()] = SSE_CONTENT_TYPE
+) -> StreamingResponse:
+    """Endpoint to handle AG-UI protocol requests and stream responses.
+
+    Args:
+        input_data: The AG-UI run input.
+        accept: The Accept header to specify the response format.
+
+    Returns:
+        A streaming response with event-stream media type.
+    """
+    return StreamingResponse(
+        agui.adapter.run(input_data, accept),
+        media_type=SSE_CONTENT_TYPE,
```

Comment:
We should be able to use `agent.to_ag_ui` in all these examples, right, instead of the `AGUIAgent` wrapper?